### PR TITLE
Bluetooth: Mesh: Fixed Coverity issues

### PR DESCRIPTION
**Pull request by Balaklaka**
Fixed issues reported by Coverity.

### CODE REVIEW

**Comment by alxelax** [comment]
@Balaklaka, it will be nice to see description\reference of issues that have been fixed by changes. Some of them do not seem reasonable to me.

File: subsys/bluetooth/mesh/sensor_srv.c [hunk 1]
```diff
@@ -394,10 +399,8 @@ static int handle_series_get(struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 		goto respond;
 	}
 
-	struct bt_mesh_sensor_column range;
-	bool ranged = (buf->len != 0);
-
 	if (buf->len == col_format->size * 2) {
+		struct bt_mesh_sensor_column range;
```
Comment (alxelax) [inline subsys/bluetooth/mesh/sensor_srv.c:403]: Seems this changs previous behavior. `range` kept value between `if` condition later `for` loop.
Now `range` value is lost between them. 

I suppose Coverity fixes shouldn't change behavior.

File: subsys/bluetooth/mesh/sensor_srv.c [hunk 2]
```diff
@@ -413,11 +416,14 @@ static int handle_series_get(struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 		}
 	} else if (buf->len != 0) {
 		/* invalid length */
+
```
Comment (alxelax) [inline subsys/bluetooth/mesh/sensor_srv.c:419]: Unintentional change.

File: subsys/bluetooth/mesh/sensor_srv.c [hunk 3]
```diff
@@ -413,11 +416,14 @@ static int handle_series_get(struct bt_mesh_model *model, struct bt_mesh_msg_ctx
...
 		LOG_WRN("Invalid length (%u)", buf->len);
 		return -EMSGSIZE;
 	}
 
 	for (uint32_t i = 0; i < sensor->series.column_count; ++i) {
+		struct bt_mesh_sensor_column range;
```
Comment (alxelax) [inline subsys/bluetooth/mesh/sensor_srv.c:425]: It is unclear what is wrong with the previous code.

**Review by alxelax — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/sensor_srv.c (hunk 1 above)
Comment (Balaklaka) [reply subsys/bluetooth/mesh/sensor_srv.c]: I will ignore the coverity warning. This is way I left ranged now, but the reason I took both. But I moved range back.
CID 113383 (#7 of 7): Branch past initialization (PW.BRANCH_PAST_INITIALIZATION)
1. branch_past_initialization: transfer of control bypasses initialization of:
 	2. name_at_decl_position: variable "ranged" (declared at line 398)

**Comment by Balaklaka** [comment]
> @Balaklaka, it will be nice to see description\reference of issues that have been fixed by changes. Some of them do not seem reasonable to me.

Reference can be found here: https://nordicsemi.atlassian.net/browse/NCSDK-22181

**Review by alxelax — APPROVED** [review]

**Review by ludvigsj — APPROVED** [review]

**Pull request merged**